2024/calcite-react/demo/src/Components/Panel/Panel.tsx

```diff
@@ -1,70 +1,61 @@
 import { useEffect, useState } from 'react';
 
-// Step 11: Panel, Shell Panel, and Block and the associated react wrappers
-import '@esri/calcite-components/dist/components/calcite-block';
-import '@esri/calcite-components/dist/components/calcite-shell-panel';
-import '@esri/calcite-components/dist/components/calcite-panel';
-
+// Step 11: Import CalciteBlock, CalciteShellPanel, and CalcitePanel
 import {
   CalciteBlock,
   CalciteShellPanel,
   CalcitePanel,
 } from '@esri/calcite-components-react';
 
-// Step 12: Import Features components
+// Step 12: Import Features component
 import '@arcgis/map-components/dist/components/arcgis-features';
 import { ArcgisFeatures } from '@arcgis/map-components-react';
 
-// Step 13: Import reactiveUtils.on to listen for view click
-import { on } from '@arcgis/core/core/reactiveUtils';
-
 interface PanelProps {
-  view: __esri.MapView | null;
+  map: HTMLArcgisMapElement | null;
   panelHeading: string;
 }
 
-export const Panel = ({ view, panelHeading }: PanelProps) => {
-  // Step 14: Set up state to store features component - to eventually set up click handle
-  const [featuresWidget, setFeaturesWidget] = useState<__esri.Features | null>(
-    null,
-  );
+export const Panel = ({ map, panelHeading }: PanelProps) => {
+  // Step 13: Set up state to store features component - to eventually set up click handle
+  const [featuresComponent, setFeaturesComponent] =
+    useState<HTMLArcgisFeaturesElement | null>(null);
 
-  // Step 15: useEffect - when features component and view is available, set up click handle
+  // Step 14: useEffect - when features component and view is available, set up click handle
   // On click, open the clicked feature if any
   useEffect(() => {
-    if (!featuresWidget || !view) return;
+    if (!featuresComponent || !map) return;
 
-    // Disable popup for features widget as feature information will be rendered in side panel
-    view.popupEnabled = false;
+    // Step 15
+    //   15a. Set view on features component
+    //   15b. disable popup for features widget as feature information will be rendered in side panel
+    featuresComponent.view = map.view;
+    map.view.popupEnabled = false;
 
-    const clickHandle = on(
-      () => view,
-      'click',
-      (event) => {
-        featuresWidget.open({
-          location: event.mapPoint,
-          fetchFeatures: true,
-        });
-      },
-    );
-    return () => clickHandle.remove();
-  }, [view, featuresWidget]);
+    const handle = map.view.on('click', (event) => {
```

Comment:
does react let you set this event listener directly on the ArcgisMap element using the on* prop?
though, I guess even if it does, it would be tricky to call the features widget from inside of there...
so I guess they way you have it is simpler